src/test/mixxxtest.h

```diff
@@ -28,6 +35,14 @@ class MixxxTest : public testing::Test {
     };
     friend class ApplicationScope;
 
+    static const QDir& testDir() {
```

Comment:
The code duplication in combination with slightly differing method names is confusing.

How about `static const QDir& getOrInitTestDir(const QString& resourcePath = QString())` that uses `ConfigObject<ConfigValue>::computeResourcePath()` if `resourcePath` is empty? Then `getTestDir()` could delegate to this function.